stl/src/mutex.cpp

```diff
@@ -123,31 +126,25 @@ static _Thrd_result mtx_do_lock(_Mtx_t mtx, const _timespec64* target) { // lock
             }
         }
 
-        if (res == WAIT_OBJECT_0 || res == WAIT_ABANDONED) {
+        if (res == WAIT_OBJECT_0) {
             if (1 < ++mtx->_Count) { // check count
                 if ((mtx->_Type & _Mtx_recursive) != _Mtx_recursive) { // not recursive, fixup count
                     --mtx->_Count;
                     res = WAIT_TIMEOUT;
                 }
             } else {
-                mtx->_Thread_id = static_cast<long>(GetCurrentThreadId());
+                mtx->_Thread_id = current_thread_id;
             }
         }
 
-        switch (res) {
-        case WAIT_OBJECT_0:
-        case WAIT_ABANDONED:
+        if (res == WAIT_OBJECT_0) {
             return _Thrd_result::_Success;
-
-        case WAIT_TIMEOUT:
+        } else { // WAIT_TIMEOUT

```

Comment:
Nitpick:
No need to `else` after the `return`.

Reply:
Also consistent with our usual conventions, although this one's debatable (we don't *strictly* avoid `else` when there's little logic afterwards).